x-pack/plugins/maps/server/maps_telemetry/maps_telemetry.ts

```diff
@@ -82,6 +94,112 @@ function getIndexPatternsWithGeoFieldCount(indexPatterns: IIndexPattern[]) {
   };
 }
 
+function getEMSLayerCount(layerLists: LayerDescriptor[][]): ILayerTypeCount[] {
+  return layerLists.map((layerList: LayerDescriptor[]) => {
+    const countsById = _(layerList).countBy((layer: LayerDescriptor) => {
+      const isEmsFile =
+        layer.sourceDescriptor !== null
+          ? layer.sourceDescriptor.type === SOURCE_TYPES.EMS_FILE
+          : false;
+      const id = (layer.sourceDescriptor as AbstractSourceDescriptor).id;
+      return isEmsFile && id ? id : 'false';
+    });
+
+    const emsCountsById = countsById.pickBy((val, key) => key !== 'false');
+    const layerTypeCount = emsCountsById.value();
+    return layerTypeCount as ILayerTypeCount;
+  }) as ILayerTypeCount[];
+}
+
+function isGeoshapeIndexPattern(
+  indexPatterns: IIndexPattern[],
+  indexPatternId: string,
+  geoField: string | undefined
+): boolean {
+  if (!geoField) {
+    return false;
+  }
+
+  const matchIndexPattern = indexPatterns.find((indexPattern: IIndexPattern) => {
+    return indexPattern.id === indexPatternId;
+  });
+
+  if (!matchIndexPattern) {
+    return false;
+  }
+
+  const fieldList: IFieldType[] =
+    matchIndexPattern.attributes && matchIndexPattern.attributes.fields
+      ? JSON.parse(matchIndexPattern.attributes.fields)
+      : [];
+
+  const matchField = fieldList.find((field: IFieldType) => {
+    return field.name === geoField;
+  });
+
+  return !!matchField && matchField.type === ES_GEO_FIELD_TYPE.GEO_SHAPE;
+}
+
+function isGeoShapeAggLayer(indexPatterns: IIndexPattern[], layer: LayerDescriptor): boolean {
+  if (layer.sourceDescriptor === null) {
+    return false;
+  }
+
+  if (
+    layer.type !== LAYER_TYPE.VECTOR &&
+    layer.type !== LAYER_TYPE.BLENDED_VECTOR &&
+    layer.type !== LAYER_TYPE.HEATMAP
+  ) {
+    return false;
+  }
+
+  const sourceDescriptor: SourceDescriptor = layer.sourceDescriptor;
+  if (sourceDescriptor.type === SOURCE_TYPES.ES_GEO_GRID) {
+    return isGeoshapeIndexPattern(
+      indexPatterns,
+      (sourceDescriptor as ESGeoGridSourceDescriptor).indexPatternId,
+      (sourceDescriptor as ESGeoGridSourceDescriptor).geoField
+    );
+  } else if (sourceDescriptor.type === SOURCE_TYPES.ES_SEARCH) {
+    const isGeoShapeIndexPattern = isGeoshapeIndexPattern(
+      indexPatterns,
+      (sourceDescriptor as ESSearchSourceDescriptor).indexPatternId,
+      (sourceDescriptor as ESSearchSourceDescriptor).geoField
+    );
+
+    return (
```

Comment:
You can avoid this extra check by moving check for scalingType === SCALING_TYPES.CLUSTERS into top level `else if`. So top level else if becomes `sourceDescriptor.type === SOURCE_TYPES.ES_SEARCH && scalingType in sourceDescriptor && sourceDescriptor. scalingType  === SCALING_TYPES.CLUSTERS`